### PR TITLE
Added syntax for the lambda

### PR DESCRIPTION
**Pull request by MakersF**
Syntax following the current language spec as in dlang.com.
I suspect it to be outdated because the current allowed syntax does not follow the spec.

### CODE REVIEW

File: DIPs/DIP1009.md [hunk 1]
```diff
@@ -38,6 +38,21 @@ change.  An existing, well-known issue or a use case for an existing projects
...
+
+```
+ValueFunctionLiteral:
+    [ CaptureParameterList_opt ] ParameterMemberAttributes FunctionLiteralBody
+    [ CaptureParameterList_opt ] Identifier => Expression
+
+CaptureParameterList:
+    CaptureParameter
+    CaptureParameter , CaptureParameterList
+
```
Comment (radcapricorn) [inline DIPs/DIP1009.md]: We should also have an AssignExpression here, i.e. for named captures (would be required for rvalues):
```d
[x = move(y)]() { return x.z }
```

**Comment by MakersF** [comment]
Indeed you are right.
I'll update after the weekend!

**Comment by radcapricorn** [comment]
I think it's safe to say there's no rush. We shouldn't be hasty, better make it solid and bulletproof. I'll also be focusing mostly on weekends.

**Review by radcapricorn — APPROVED** [review]
Lookin' good

**Pull request merged**